Design-concepts.md

```diff
@@ -260,6 +264,554 @@ For FuseML to be functional, the following design elements and components are ne
+
+The example described in this section is minimal. It assumes that in the experimentation phase, the data scientist is creating a monolith: a single logical block of code (represented by a single MLFlow codeset) that bundles together all the steps that make up the experimentation part of the pipeline required to train a machine learning model: data preparation, model definition, training and validation. However, theoretically, the FuseML design and its pipelines are flexible enough to allow data scientists to split these monoliths into modules that can be further orchestrated as individual, composable and reusable pipeline components.
+
+The following high-level composable pipeline segment templates are featured in this exercise:
+* builder segment: takes in an MLFlow compatible codeset artifact as input and generates a runnable artifact representing the environment that can be used to run any of its entry points. This is based on the `mlflow-builder` runnable installed by the MLFlow extension
+* trainer segment: takes in an MLFlow compatible codeset artifact as input, runs it and outputs the resulted trained ML model. This segment is based on the runnable generated at runtime by the builder segment
+* predictor segment: takes in a trained ML model as input, starts a prediction service for it and outputs its exposed endpoint (i.e. the URL where the service receives HTTP calls for prediction). This is based on the `kfserving-predictor` runnable installed by the KFServing extension
+
+Each of these segments can be instantiated and executed independently, as long as there are available inputs (e.g. compatible codesets and models) that they can consume and as long as the values of all other input parameters are explicitly specified (or have default values). The steps detailed in the next part will cover defining the individual pipeline templates and then building a master end-to-end pipeline composed of all three segments which is automatically run according to the available MLFlow compatible codeset versions published as detailed in the _Publishing Codesets_ section. 
+
+##### Inline Pipeline Templates
+
+Before going into how runnables can be used as building blocks for pipelines, here's what the complete end-to-end pipeline may look like _without_ the use of runnables as composable components:
+
+  ```
+  pipeline-template:
+    name: mlflow-kfserving-e2e
+    description: |
+      End-to-end pipeline template that takes in an MLFlow compatible codeset,
+      runs the MLFlow project to train a model, then creates a KFServing prediction 
+      service that can be used to run predictions against the model.
+    inputs:
+      - name: mlflow-codeset
+        description: an MLFlow compatible codeset
+        type: codeset
+        labels:
+          - mlflow-project
+      - name: predictor
+        description: type of predictor engine
+        type: string
+        default: auto
```

Comment:
I think I understand what you're saying and here's why having a `value` field here doesn't make sense...

This operation and example yaml file describes defining a pipeline _template_, which lists the input parameters that it can accept, not their actual values (minus the occasional default value for optional parameters). The parameter values are filled in during a different operation: instantiating a template. During the instantiation of a template, the end user references a template and fills in the input parameter values, resulting in a fully defined pipeline that can be executed.